Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialTypePreBuilder.cpp

```diff
@@ -0,0 +1,274 @@
+                    for (const AZStd::string& path : possibleDependencies)
+                    {
+                        response.m_sourceFileDependencyList.push_back({});
+                        response.m_sourceFileDependencyList.back().m_sourceFileDependencyPath = path;
+                    }
+
+                    // We don't need to add m_azsli as a dependency because that will be #included into the final azsl file, so the shader asset builder
+                    // will account for that dependency.
+                }
+            }
+
+            response.m_result = AssetBuilderSDK::CreateJobsResultCode::Success;
+        }
+
+        void MaterialTypePreBuilder::ProcessJob(const AssetBuilderSDK::ProcessJobRequest& request, AssetBuilderSDK::ProcessJobResponse& response) const
+        {
+            AssetBuilderSDK::JobCancelListener jobCancelListener(request.m_jobId);
+
+            if (jobCancelListener.IsCancelled())
+            {
+                response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Cancelled;
+                return;
+            }
+
+            if (m_isShuttingDown)
+            {
+                response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Cancelled;
+                return;
+            }
+
+            AZStd::string materialTypeName;
+            AZ::StringFunc::Path::GetFileName(request.m_sourceFile.c_str(), materialTypeName);
+
+            AZ::u32 nextProductSubID = 0;
+
+            auto materialTypeLoadResult = MaterialUtils::LoadMaterialTypeSourceData(request.m_fullPath);
+            if (!materialTypeLoadResult.IsSuccess())
+            {
+                AZ_Error(MaterialTypePreBuilderName, false, "Failed to load material type file.");
+                return;
+            }
+            MaterialTypeSourceData materialType = materialTypeLoadResult.TakeValue();
+            materialType.m_shaderCollection.clear();
+
+            // Generate the required shaders
+            for (const auto& [materialPipelineFilePath, materialPipeline] : m_materialPipelines)
+            {
+                AZStd::string materialPipelineName;
+                AZ::StringFunc::Path::GetFileName(materialPipelineFilePath.c_str(), materialPipelineName);
+
+
+                // TODO: Eventually we'll use a script and inputs from the material type to decide which shader templates need to be used
+                for (const MaterialPipelineSourceData::ShaderTemplate& shaderTemplate : materialPipeline.m_shaderTemplates)
+                {
+                    AZStd::string shaderName;
+                    AZ::StringFunc::Path::GetFileName(shaderTemplate.m_shader.c_str(), shaderName); // This will remove the ".template" extension
+                    AZ::StringFunc::Path::ReplaceExtension(shaderName, "");                         // This will remove the ".shader" extension
```

Comment:
`AZ::IO::Path` should be used over the `AZ::StringFunc::Path` API. 
I have made a (jira) ticket to remove the AZ::StringFunc::Path API one day